### DISCUSSION TITLE
EVTV ESP32 CANDue & savvycan

### DISCUSSION THREAD

**Discussion by x13pow** [collin80/SavvyCAN#676] [Q&A]
Good Morning All, 

I wanted to ask that if i purchase a EVTV ESP32 CANDue do i need to program it first or does it work with savvycan out the box? I also wanted to know if you would need to reprogram the EVTV ESP32 CANDue if you needed a different speed ? 

I only want the device to be able to sniff CANbus so look at some data which i was waiting to get hold of a macchina m2 but i do not thing thats going to happen as they are never in stock. If someone has any better suggestions on hardware please post away :) 

Thank you in advance

**Comment by danjhunter** [comment]
@x13pow EVTV ESP32 CANDue needs to have firmware installed first. I just bought a board and I am in the process of getting it to connect to SavvyCAN. The need to reprogram the EVTV ESP32 CANDue for different speeds would be a question for @collin80 ?

**Comment by boroko** [comment]
I too am having a bit of trouble getting this right on an EVTV ESP32 CANdue and Windows.  EVTV's site isn't clear that it needs to be loaded with Collins code and neither is does the GIThub page.  I've read most of the discussions here looking for tidbits, and I'm slowly piecing it together.  I'm reasonably comfortable with Arduino, but I have to admit it would be nicer if it were spelled out a bit clearer.  There are a lot of files in Collins repo, and a bit more hand holding would be helpful to me, and probably others.  From everything I've seen, I am very impressed with SavvyCAN, I'm just a little frustrated that I feel like I'm reverse engineering mode just to get to reverse engineering some CANbus.   This is not meant as a venting session as much as a plea to summarize a set of simple instructions in one place for us simple folk.  Thanks

**Reply by danjhunter** [reply]
@boroko 1.) Have you successfully loaded the firmware onto the EVTV ESP32 CANdue board? 2.) Have you installed the latest version of SavvyCAN V213 for Windows? https://github.com/collin80/SavvyCAN/releases/tag/V213

**Reply by boroko** [reply]
I do have V213 loaded on windows.  I'm working on getting the firmware loaded onto the EVTV ESP32...   I began to suspect the firmware wasn't on the ESP32, and I'm going down the path that it isn't.  Just got Arduino_ESP32 loaded into the board manager and the partition set to Minimal SPIFFS.  I'm running Arduino 2.1.1 and started  blank sketch with and include of esp_can and can_common.    Not exactly sure what to do next, so I was going to monkey-poke for a while.  Probably could use some advice at this point.

**Reply by boroko** [reply]
I was confused for a while seeing the LED activity on the EVTV.  Thought the firmware might be on it already and I was cautious about overwriting it and messing it up.  I'm now guessing that was just USB activity, and the lack of firmware explains the inability to connect to it with SavvyCAN.

**Reply by boroko** [reply]
Another detail once I figure out what to load:  I'm selecting a ESP32-WROOM-DA Module as the board, seeing nothing better.  Also seeing 2 USB ports, which I suspect is correct.

**Comment by collin80** [comment]
There is a binary updater that can flash the proper firmware for you: https://www.savvycan.com/ESP32RET_Updater.zip

That's a lot easier than trying to compile it yourself. But, when you do compile, you want "ESP32 Dev Module" as the board type and "Minimal SPIFFS" as the partition scheme.

**Comment by boroko** [comment]
I have seen that, and in fact am trying to understand from that page.  How do I run the updater?  Is it just loaded through the IDE somehow, of does it need to be in a sketch?

**Comment by x13pow** [comment]
No you need to run the batch file ! Do not even need IDE installed

Kind Regards,
Matt Powell
Technical Director
Mobile: 07879440597
Email: ***@***.***
Phone: 01689 879000 | Fax: 020 83025706 
www.powell-systems.co.uk
                                                                                                                                                                                                                                                                                                          
​​Trend Outstanding Partner Performance Award 2020 
​Trend Outstanding Partner Performance Award 2019​
​Trend Outstanding Partner Performance Award 2018 
​Trend Outstanding Partner Performance Award 2017
​Trend Outstanding Partner Performance Award 2016 
​Finalist of BCIA 2016 ‘Best Maintenance Provider’ 
McDonalds Supplier Of The Year 2015 
​Trend Outstanding Partner Performance Award 2015 


 Want to save money on BeMS Maintenance? Click Here for a quote 
                                                                                                                                                                                                                                                                                                          
Powell System Engineering Ltd • Unit 2, Fitzroy Business Park • Sandy Lane • Sidcup • DA14 5NL • Registered in England number 2800445 
This message and its contents are strictly confidential. It is intended for the named recipient(s) only and should not be copied, distributed or relied upon by those for whom it is not intended. Powell Systems Engineering Ltd holds no responsibility for the information contained in this message as the opinion stated is that of the sender unless clearly stated otherwise. 

On 29 Oct 2023, at 22:27, boroko ***@***.***> wrote:

﻿

I have seen that, and in fact am trying to understand from that page. How do I run the updater? Is it just loaded through the IDE somehow, of does it need to be in a sketch?

—
Reply to this email directly, view it on GitHub<https://github.com/collin80/SavvyCAN/discussions/676#discussioncomment-7418098>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJKHSNX5BW3JJZWGLVGLA5DYB3J3TAVCNFSM6AAAAAA5SHF2Z6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMJYGA4TQ>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by collin80** [comment]
There's a batch file. It has all the executables and binaries already there. You just unzip it and run the batch file (obviously with the device plugged in). It should successfully flash on the first COM port then not find anything to flash on the second one. That's all normal.

**Comment by boroko** [comment]
I appreciate the answers, but on [https://github.com/collin80/ESP32RET](url) I see nothing that looks like a batch file.  Closest thing I could guess is ESP32RET.ino, and I was going to try and load that.  I can be thick at times, but I've been chewing on this for a few weeks.

**Comment by boroko** [comment]
Ok, maybe the thickness is wearing off.  I unzipped the updater instead of looking at the files on that link, and I think I've found the batch files.....

**Comment by x13pow** [comment]
In the link Colin gave a few messages ago there is a zip file. If you download that you will see that there is a update.bat
Sent from my iPhone

Kind Regards,
Matt Powell
Technical Director
Mobile: 07879440597
Email: ***@***.***
Phone: 01689 879000 | Fax: 020 83025706 
www.powell-systems.co.uk
                                                                                                                                                                                                                                                                                                          
​​Trend Outstanding Partner Performance Award 2020 
​Trend Outstanding Partner Performance Award 2019​
​Trend Outstanding Partner Performance Award 2018 
​Trend Outstanding Partner Performance Award 2017
​Trend Outstanding Partner Performance Award 2016 
​Finalist of BCIA 2016 ‘Best Maintenance Provider’ 
McDonalds Supplier Of The Year 2015 
​Trend Outstanding Partner Performance Award 2015 


 Want to save money on BeMS Maintenance? Click Here for a quote 
                                                                                                                                                                                                                                                                                                          
Powell System Engineering Ltd • Unit 2, Fitzroy Business Park • Sandy Lane • Sidcup • DA14 5NL • Registered in England number 2800445 
This message and its contents are strictly confidential. It is intended for the named recipient(s) only and should not be copied, distributed or relied upon by those for whom it is not intended. Powell Systems Engineering Ltd holds no responsibility for the information contained in this message as the opinion stated is that of the sender unless clearly stated otherwise. 

On 29 Oct 2023, at 22:37, boroko ***@***.***> wrote:

﻿

I appreciate the answers, but on https://github.com/collin80/ESP32RET<url> I see nothing that looks like a batch file. Closest thing I could guess is ESP32RET.ino, and I was going to try and load that. I can be thick at times, but I've been chewing on this for a few weeks.

—
Reply to this email directly, view it on GitHub<https://github.com/collin80/SavvyCAN/discussions/676#discussioncomment-7418126>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJKHSNROSOKJYCODNEGTDHLYB3LBFAVCNFSM6AAAAAA5SHF2Z6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMJYGEZDM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by boroko** [comment]
The problem I see now was that I was looking at [https://github.com/collin80/ESP32RET](url) instead of [https://www.savvycan.com/ESP32RET_Updater.zip](url).  I appreciate the help guys.  
btw, x13pow: That has to be the longest signature I have ever experienced :-)

**Comment by boroko** [comment]
UPDATE:  I am now the proud owner of a EVTV ESP32_CANdue with a CONNECTED indication on SavvyCAN.  Very appreciated.

**Comment by boroko** [comment]
Been thinking about it, and looking back on the SavvyCAN main page trying to learn from what I overlooked.  
Maybe an slightly expanded entry on the 
"[ESP32RET Firmware Updater](https://savvycan.com/ESP32RET_Updater.zip) (Updated August 2, 2023 - Version 618)" 
area would have caught my attention and saved your time explaining.  Possibly clarifying: "Firmware updater for XXX hardware: Unzip and run "updater.bat" to install the complete firmware onto your hardware.  No need to compile and install through the Arduino IDE."  
Might make it easier for dolts like me ;-)
